### DISCUSSION TITLE
Passing compiler flags to libraries downloaded through vcpkg

### DISCUSSION THREAD

**Discussion by LittleTreeTrunk** [microsoft/vcpkg#39556] [Q&A]
We want use SQLite in our project. We are using vcpkg in Manifest mode.  We are using the default vcpkg registry. We are specifying the library dependencies in vcpkg.json file. But, we wish to pass additional preprocessor directive to the compiler at the time of SQLite build (eg. SQLITE_ENABLE_UNLOCK_NOTIFY). What is the right way to do this?

**Comment by duckdoom5** [comment]
You could set up a custom triplet file (super simple, just a few lines of code). And then specify the custom triplet overlay file in your vcpkg.json or in CMAKE with a variable.

See this documentation link for more info: https://learn.microsoft.com/en-us/vcpkg/users/examples/overlay-triplets-linux-dynamic

Then you can specify your compiler flags in the triplet file. (See https://learn.microsoft.com/en-us/vcpkg/users/triplets?source=recommendations#vcpkg_cxx_flags)
